src/WeakCss.elm

```diff
@@ -181,13 +187,47 @@ nestMany listToAdd =
 -}
 withStates : List ( String, Bool ) -> ClassName -> Attribute msg
 withStates states =
-    class << toStringWithStates states
+    let
+        activeStates : List String
+        activeStates =
+            states
+                |> List.filterMap
+                    (\( string, state ) ->
+                        if state then
+                            Just string
+
+                        else
+                            Nothing
+                    )
+    in
+    Attrs.class << toStringWithStates activeStates
+
+
+{-| Add state to last element [`ClassName`](#ClassName) and convert to `Html.Attribute msg`.
+
+    import Html.Attributes exposing (class)
+
+    namespace "menu"
+        |> add "item"
+        |> withStates_ [ "active", "highlighted" ]
+    --> class "menu__item active highlighted"
+
+    namespace "menu"
+        |> withStates_ []
+    --> class "menu"
+
+-}
+withStates_ : List String -> ClassName -> Attribute msg
+withStates_ states =
```

Comment:
I was thinking about some better name, like `withActiveStates` but not sure about it 🤔 We can go probably with this for now.

Reply:
I’d be in favour of renaming this function. This is public package, so I’d rather have better API and names for it. A D the name with underscore doesn’t look professionally :)